# Checkpoint Three: Cleaning Data

Now you are ready to clean your data. Before starting coding, provide the link to your dataset below.

My dataset:

Import the necessary libraries and create your dataframe(s).

In [584]:
import pandas as pd
import numpy as np 

In [585]:
former_names = pd.read_csv('former_names.csv')
goal_scorers = pd.read_csv('goalscorers.csv')
results = pd.read_csv('results.csv')
shootouts = pd.read_csv('shootouts.csv')
continents = pd.read_csv('continents2.csv')

I want to add continents into the results dataframe, and I also want to update all country names to have their current names

## Irregular Data

Detect outliers in your dataset and handle them as needed. Use code comments to make notes about your thought process.

In [586]:
continents = continents.drop(columns=["alpha-2", "alpha-3", "country-code", "iso_3166-2", "intermediate-region", "region-code", "sub-region-code", "intermediate-region-code"])
continents.head()
continents.value_counts('region')

region
Africa      60
Americas    57
Asia        51
Europe      51
Oceania     29
Name: count, dtype: int64

In [587]:
continents.head()
continents.value_counts('sub-region')

sub-region
Sub-Saharan Africa                 53
Latin America and the Caribbean    52
Western Asia                       18
Southern Europe                    16
Northern Europe                    16
South-eastern Asia                 11
Eastern Europe                     10
Polynesia                          10
Western Europe                      9
Southern Asia                       9
Eastern Asia                        8
Micronesia                          8
Northern Africa                     7
Australia and New Zealand           6
Central Asia                        5
Melanesia                           5
Northern America                    5
Name: count, dtype: int64

country names will need to be updated across all dataframes for consistency

In [588]:
former_names

current                                former  start_date  \
0                 Benin                               Dahomey  1959-11-08   
1          Burkina Faso                           Upper Volta  1960-04-14   
2               Curaçao                  Netherlands Antilles  1957-03-03   
3        Czechoslovakia                               Bohemia  1903-04-05   
4        Czechoslovakia                   Bohemia and Moravia  1939-01-01   
5        Czechoslovakia  Representation of Czechs and Slovaks  1993-03-24   
6              DR Congo                         Belgian Congo  1948-05-25   
7              DR Congo                    Congo-Léopoldville  1963-04-12   
8              DR Congo                        Congo-Kinshasa  1965-01-09   
9              DR Congo                                 Zaïre  1971-01-10   
10             Djibouti                     French Somaliland  1947-12-05   
11                Egypt                  United Arab Republic  1958-02-22   
12             Eswatini                             Swaziland  1968-05-01   
13                Ghana                            Gold Coast  1950-05-28   
14        Guinea-Bissau                     Portuguese Guinea  1953-06-02   
15               Guyana                        British Guiana  1905-07-21   
16            Indonesia                     Dutch East Indies  1934-05-13   
17               Israel                   Mandatory Palestine  1934-03-16   
18               Malawi                             Nyasaland  1957-08-28   
19             Malaysia                                Malaya  1948-06-20   
20              Myanmar                                 Burma  1952-03-14   
21      North Macedonia                             Macedonia  1993-10-13   
22     Northern Ireland                               Ireland  1882-02-18   
23  Republic of Ireland                      Irish Free State  1924-05-28   
24  Republic of Ireland                                  Éire  1937-01-01   
25               Russia                          Soviet Union  1924-11-16   
26               Russia                                   CIS  1992-01-25   
27                Samoa                         Western Samoa  1979-08-31   
28            Sri Lanka                                Ceylon  1952-03-24   
29             Suriname                          Dutch Guyana  1921-01-28   
30             Tanzania                            Tanganyika  1945-01-01   
31              Vanuatu                          New Hebrides  1951-10-04   
32               Zambia                     Northern Rhodesia  1947-02-22   
33             Zimbabwe                     Southern Rhodesia  1946-06-16   

      end_date  
0   1975-11-30  
1   1984-08-04  
2   2010-10-10  
3   1919-01-01  
4   1945-05-01  
5   1993-11-17  
6   1956-01-02  
7   1964-07-19  
8   1970-11-24  
9   1997-04-27  
10  1977-06-27  
11  1971-09-11  
12  2018-04-19  
13  1957-03-06  
14  1973-09-24  
15  1967-08-02  
16  1951-03-04  
17  1940-04-27  
18  1964-07-04  
19  1963-08-16  
20  1989-01-01  
21  2018-11-19  
22  1956-04-11  
23  1936-12-06  
24  1953-03-25  
25  1991-11-13  
26  1992-06-18  
27  1996-11-15  
28  1972-05-22  
29  1975-11-25  
30  1964-04-26  
31  1980-07-30  
32  1964-10-25  
33  1980-04-18

In [589]:
former_names = former_names.drop(['start_date', 'end_date'], axis=1)

In [590]:
former_names.head()
#I think for thsi analysis I will be doing for this project, the start and end dat of former country names won't he helpful or relevant. I think it would be really interesting to look at national performance based on socio-political activity (presumedly could be gleaned from changes in country names) but that is not where I intend to dedicate my depth of analysis. I will drop the start_date and end_date columns 

current                former
0           Benin               Dahomey
1    Burkina Faso           Upper Volta
2         Curaçao  Netherlands Antilles
3  Czechoslovakia               Bohemia
4  Czechoslovakia   Bohemia and Moravia

need to create a dictionary with the former and current names

In [591]:
names_dictionary = former_names.set_index(['former']).to_dict()['current']
names_dictionary

{'Dahomey': 'Benin',
 'Upper Volta': 'Burkina Faso',
 'Netherlands Antilles': 'Curaçao',
 'Bohemia': 'Czechoslovakia',
 'Bohemia and Moravia': 'Czechoslovakia',
 'Representation of Czechs and Slovaks': 'Czechoslovakia',
 'Belgian Congo': 'DR Congo',
 'Congo-Léopoldville': 'DR Congo',
 'Congo-Kinshasa': 'DR Congo',
 'Zaïre': 'DR Congo',
 'French Somaliland': 'Djibouti',
 'United Arab Republic': 'Egypt',
 'Swaziland': 'Eswatini',
 'Gold Coast': 'Ghana',
 'Portuguese Guinea': 'Guinea-Bissau',
 'British Guiana': 'Guyana',
 'Dutch East Indies': 'Indonesia',
 'Mandatory Palestine': 'Israel',
 'Nyasaland': 'Malawi',
 'Malaya': 'Malaysia',
 'Burma': 'Myanmar',
 'Macedonia': 'North Macedonia',
 'Ireland': 'Northern Ireland',
 'Irish Free State': 'Republic of Ireland',
 'Éire': 'Republic of Ireland',
 'Soviet Union': 'Russia',
 'CIS': 'Russia',
 'Western Samoa': 'Samoa',
 'Ceylon': 'Sri Lanka',
 'Dutch Guyana': 'Suriname',
 'Tanganyika': 'Tanzania',
 'New Hebrides': 'Vanuatu',
 'Northern Rhodesi

In [592]:
results.head()

date home_team away_team  home_score  away_score tournament     city  \
0  1872-11-30  Scotland   England         0.0         0.0   Friendly  Glasgow   
1  1873-03-08   England  Scotland         4.0         2.0   Friendly   London   
2  1874-03-07  Scotland   England         2.0         1.0   Friendly  Glasgow   
3  1875-03-06   England  Scotland         2.0         2.0   Friendly   London   
4  1876-03-04  Scotland   England         3.0         0.0   Friendly  Glasgow   

    country  neutral  
0  Scotland    False  
1   England    False  
2  Scotland    False  
3   England    False  
4  Scotland    False

Since I generated a dictionary with the former and current country names, I can use .replace() in the country column very easily. I will need to do this with all dataframes that list the country/team





Then we can create a dictionary of country names and continents, and insert a column of continents into each dataframe

In [593]:
continents.head()

name   region       sub-region
0     Afghanistan     Asia    Southern Asia
1   Åland Islands   Europe  Northern Europe
2         Albania   Europe  Southern Europe
3         Algeria   Africa  Northern Africa
4  American Samoa  Oceania        Polynesia

In [594]:
continents_dictionary = continents.set_index(['name']).to_dict()['region']
continents_dictionary


{'Afghanistan': 'Asia',
 'Åland Islands': 'Europe',
 'Albania': 'Europe',
 'Algeria': 'Africa',
 'American Samoa': 'Oceania',
 'Andorra': 'Europe',
 'Angola': 'Africa',
 'Anguilla': 'Americas',
 'Antarctica': nan,
 'Antigua and Barbuda': 'Americas',
 'Argentina': 'Americas',
 'Armenia': 'Asia',
 'Aruba': 'Americas',
 'Australia': 'Oceania',
 'Austria': 'Europe',
 'Azerbaijan': 'Asia',
 'Bahamas': 'Americas',
 'Bahrain': 'Asia',
 'Bangladesh': 'Asia',
 'Barbados': 'Americas',
 'Belarus': 'Europe',
 'Belgium': 'Europe',
 'Belize': 'Americas',
 'Benin': 'Africa',
 'Bermuda': 'Americas',
 'Bhutan': 'Asia',
 'Bolivia': 'Americas',
 'Bonaire, Sint Eustatius and Saba': 'Americas',
 'Bosnia And Herzegovina': 'Europe',
 'Botswana': 'Africa',
 'Bouvet Island': 'Americas',
 'Brazil': 'Americas',
 'British Indian Ocean Territory': 'Africa',
 'Brunei Darussalam': 'Asia',
 'Bulgaria': 'Europe',
 'Burkina Faso': 'Africa',
 'Burundi': 'Africa',
 'Cabo Verde': 'Africa',
 'Cambodia': 'Asia',
 'Cameroon'

In [595]:
results["country"] = results["country"].replace(names_dictionary)

In [596]:
results['continent'] = results['country'].map(continents_dictionary)
results


date              home_team        away_team  home_score  \
0      1872-11-30               Scotland          England         0.0   
1      1873-03-08                England         Scotland         4.0   
2      1874-03-07               Scotland          England         2.0   
3      1875-03-06                England         Scotland         2.0   
4      1876-03-04               Scotland          England         3.0   
...           ...                    ...              ...         ...   
48845  2025-11-18  Saint Kitts and Nevis           Belize         2.0   
48846  2025-11-18    Antigua and Barbuda           Guyana         1.0   
48847  2025-11-18                   Iran       Uzbekistan         0.0   
48848  2025-11-19                   Laos          Vietnam         0.0   
48849  2025-11-20                Vanuatu  Solomon Islands         NaN   

       away_score                   tournament          city  \
0             0.0                     Friendly       Glasgow   
1             2.0                     Friendly        London   
2             1.0                     Friendly       Glasgow   
3             2.0                     Friendly        London   
4             0.0                     Friendly       Glasgow   
...           ...                          ...           ...   
48845         6.0              CONCACAF Series    Basseterre   
48846         4.0              CONCACAF Series   North Sound   
48847         0.0     Al Ain International Cup        Al Ain   
48848         2.0  AFC Asian Cup qualification     Vientiane   
48849         NaN     MSG Prime Minister's Cup  Port Moresby   

                     country  neutral continent  
0                   Scotland    False       NaN  
1                    England    False       NaN  
2                   Scotland    False       NaN  
3                    England    False       NaN  
4                   Scotland    False       NaN  
...                      ...      ...       ...  
48845  Saint Kitts and Nevis    False  Americas  
48846    Antigua and Barbuda    False  Americas  
48847   United Arab Emirates     True      Asia  
48848                   Laos    False      Asia  
48849       Papua New Guinea     True   Oceania  

[48850 rows x 10 columns]

in the continent dataframe/dictionary, Scotland, England, Wales, and Northern Ireland are all listed as United Kingdowm for their country. I am going to create a mini dictionary listing each of those in Europe, as their individual national identity is an important part of this analysis

In [597]:
united_kingdom = {'England': "Europe", 'Scotland': "Europe", 'Wales': "Europe", 'Northern Ireland': "Europe"}
united_kingdom

{'England': 'Europe',
 'Scotland': 'Europe',
 'Wales': 'Europe',
 'Northern Ireland': 'Europe'}

In [598]:
results["continent"] = results["continent"].fillna(results["country"].map(united_kingdom))
results.head()

date home_team away_team  home_score  away_score tournament     city  \
0  1872-11-30  Scotland   England         0.0         0.0   Friendly  Glasgow   
1  1873-03-08   England  Scotland         4.0         2.0   Friendly   London   
2  1874-03-07  Scotland   England         2.0         1.0   Friendly  Glasgow   
3  1875-03-06   England  Scotland         2.0         2.0   Friendly   London   
4  1876-03-04  Scotland   England         3.0         0.0   Friendly  Glasgow   

    country  neutral continent  
0  Scotland    False    Europe  
1   England    False    Europe  
2  Scotland    False    Europe  
3   England    False    Europe  
4  Scotland    False    Europe

In [599]:
goal_scorers.head()
goal_scorers["team"] = goal_scorers["team"].replace(names_dictionary)
goal_scorers['continent'] = goal_scorers['team'].map(continents_dictionary)
goal_scorers["continent"] = goal_scorers["continent"].fillna(goal_scorers["team"].map(united_kingdom))
goal_scorers.head()


date  home_team away_team       team            scorer  minute  \
0  1916-07-02      Chile   Uruguay    Uruguay   José Piendibene    44.0   
1  1916-07-02      Chile   Uruguay    Uruguay  Isabelino Gradín    55.0   
2  1916-07-02      Chile   Uruguay    Uruguay  Isabelino Gradín    70.0   
3  1916-07-02      Chile   Uruguay    Uruguay   José Piendibene    75.0   
4  1916-07-06  Argentina     Chile  Argentina     Alberto Ohaco     2.0   

   own_goal  penalty continent  
0     False    False  Americas  
1     False    False  Americas  
2     False    False  Americas  
3     False    False  Americas  
4     False    False  Americas

In [600]:
shootouts.head()
shootouts["away_team"] = shootouts["away_team"].replace(names_dictionary)
shootouts["winner"] = shootouts["winner"].replace(names_dictionary)
shootouts["winner_continent"] = shootouts['winner'].map(continents_dictionary)
shootouts["away_continent"] = shootouts['away_team'].map(continents_dictionary)
shootouts["winner_continent"] = shootouts["winner_continent"].fillna(shootouts["winner"].map(united_kingdom))
shootouts["away_continent"] = shootouts["away_continent"].fillna(shootouts["away_team"].map(united_kingdom))
shootouts.head()


date    home_team         away_team       winner first_shooter  \
0  1967-08-22        India            Taiwan       Taiwan           NaN   
1  1971-11-14  South Korea  Vietnam Republic  South Korea           NaN   
2  1972-05-07  South Korea              Iraq         Iraq           NaN   
3  1972-05-17     Thailand       South Korea  South Korea           NaN   
4  1972-05-19     Thailand          Cambodia     Thailand           NaN   

  winner_continent away_continent  
0             Asia           Asia  
1             Asia            NaN  
2             Asia           Asia  
3             Asia           Asia  
4             Asia           Asia

## Missing Data

Test your dataset for missing data and handle it as needed. Make notes in the form of code comments as to your thought process.

In [601]:
results.isnull().value_counts()

date   home_team  away_team  home_score  away_score  tournament  city   country  neutral  continent
False  False      False      False       False       False       False  False    False    False        46063
                                                                                          True          2786
                             True        True        False       False  False    False    False            1
Name: count, dtype: int64

In [602]:
results = results.drop([48849])
#I was able to see that index [48849] was the only row with null values, so I am going to manually drop it since it is the only null value 


In [603]:
results.isnull().value_counts()


date   home_team  away_team  home_score  away_score  tournament  city   country  neutral  continent
False  False      False      False       False       False       False  False    False    False        46063
                                                                                          True          2786
Name: count, dtype: int64

Even with the data cleaning above with country name and continent reconciliation, there are still 2786 countries that fall outside of those parameters. This is 6% of the total data set. For consistency I will drop these rows. 

In [604]:
results = results.dropna()
results.isnull().value_counts()

date   home_team  away_team  home_score  away_score  tournament  city   country  neutral  continent
False  False      False      False       False       False       False  False    False    False        46063
Name: count, dtype: int64

In [605]:
shootouts.isnull().value_counts()
#there are a considerable amount of rows where the first_shooter is null. I do not plan on including the first shooter in a shootout in my analysis, so I am going to drop that column,. 

date   home_team  away_team  winner  first_shooter  winner_continent  away_continent
False  False      False      False   True           False             False             351
                                     False          False             False             214
                                     True           True              True               47
                                     False          True              True               12
                                     True           False             True               11
                                                    True              False              11
                                     False          False             True                6
                                                    True              False               5
Name: count, dtype: int64

In [606]:
shootouts.head()


date    home_team         away_team       winner first_shooter  \
0  1967-08-22        India            Taiwan       Taiwan           NaN   
1  1971-11-14  South Korea  Vietnam Republic  South Korea           NaN   
2  1972-05-07  South Korea              Iraq         Iraq           NaN   
3  1972-05-17     Thailand       South Korea  South Korea           NaN   
4  1972-05-19     Thailand          Cambodia     Thailand           NaN   

  winner_continent away_continent  
0             Asia           Asia  
1             Asia            NaN  
2             Asia           Asia  
3             Asia           Asia  
4             Asia           Asia

In [607]:
shootouts = shootouts.drop(columns=["first_shooter"])


In [608]:
goal_scorers.isnull().value_counts()
# the scorer itself is the main "point" of this particular dataset, so with a null value in that field there is not much I can do with that particular row. Since the number of rows with no scorer are negligible, I am goign to drop the rows where [scorer] is null.
# the [minute] column, which captures the minute of the goal score will not play a role in my analysis. However, there may be records where there is a goal scorer but no minute recorded. In this case I would still want to keep the rows with scorers. Dropping rows with null scorers is enough to remedy the missing data here

date   home_team  away_team  team   scorer  minute  own_goal  penalty  continent
False  False      False      False  False   False   False     False    False        41143
                                                                       True          3048
                                            True    False     False    False          179
                                    True    True    False     False    False           30
                                    False   True    False     False    True            29
                                    True    True    False     False    True            18
Name: count, dtype: int64

In [609]:
goal_scorers = goal_scorers.dropna(subset=['scorer'])

## Unnecessary Data

Look for the different types of unnecessary data in your dataset and address it as needed. Make sure to use code comments to illustrate your thought process.

In [610]:
goal_scorers.head()
#I am actually going to go ahead and drop the minute column all together - this datapoint will not have a role in my analysis
#I also don't think I am going to consider own goal or penalty in my analysis. There are so many things I could analyze in this project, but I don't want to get too wide and compromise depth.

date  home_team away_team       team            scorer  minute  \
0  1916-07-02      Chile   Uruguay    Uruguay   José Piendibene    44.0   
1  1916-07-02      Chile   Uruguay    Uruguay  Isabelino Gradín    55.0   
2  1916-07-02      Chile   Uruguay    Uruguay  Isabelino Gradín    70.0   
3  1916-07-02      Chile   Uruguay    Uruguay   José Piendibene    75.0   
4  1916-07-06  Argentina     Chile  Argentina     Alberto Ohaco     2.0   

   own_goal  penalty continent  
0     False    False  Americas  
1     False    False  Americas  
2     False    False  Americas  
3     False    False  Americas  
4     False    False  Americas

In [611]:
goal_scorers = goal_scorers.drop(['minute', 'own_goal', 'penalty'], axis=1)

In [612]:
results.head()
#I am not sure what the neutral column is for, and I know it will not play a role in my analysis, so I am going to drop that column. 

date home_team away_team  home_score  away_score tournament     city  \
0  1872-11-30  Scotland   England         0.0         0.0   Friendly  Glasgow   
1  1873-03-08   England  Scotland         4.0         2.0   Friendly   London   
2  1874-03-07  Scotland   England         2.0         1.0   Friendly  Glasgow   
3  1875-03-06   England  Scotland         2.0         2.0   Friendly   London   
4  1876-03-04  Scotland   England         3.0         0.0   Friendly  Glasgow   

    country  neutral continent  
0  Scotland    False    Europe  
1   England    False    Europe  
2  Scotland    False    Europe  
3   England    False    Europe  
4  Scotland    False    Europe

In [613]:
results = results.drop(['neutral'], axis=1)

In [614]:
shootouts.head()

date    home_team         away_team       winner winner_continent  \
0  1967-08-22        India            Taiwan       Taiwan             Asia   
1  1971-11-14  South Korea  Vietnam Republic  South Korea             Asia   
2  1972-05-07  South Korea              Iraq         Iraq             Asia   
3  1972-05-17     Thailand       South Korea  South Korea             Asia   
4  1972-05-19     Thailand          Cambodia     Thailand             Asia   

  away_continent  
0           Asia  
1            NaN  
2           Asia  
3           Asia  
4           Asia

## Inconsistent Data

Check for inconsistent data and address any that arises. As always, use code comments to illustrate your thought process.

In [626]:
goal_scorers.head()

date  home_team away_team       team            scorer continent
0  1916-07-02      Chile   Uruguay    Uruguay   José Piendibene  Americas
1  1916-07-02      Chile   Uruguay    Uruguay  Isabelino Gradín  Americas
2  1916-07-02      Chile   Uruguay    Uruguay  Isabelino Gradín  Americas
3  1916-07-02      Chile   Uruguay    Uruguay   José Piendibene  Americas
4  1916-07-06  Argentina     Chile  Argentina     Alberto Ohaco  Americas

In [624]:
results.head()

date home_team away_team  home_score  away_score tournament     city  \
0  1872-11-30  Scotland   England         0.0         0.0   Friendly  Glasgow   
1  1873-03-08   England  Scotland         4.0         2.0   Friendly   London   
2  1874-03-07  Scotland   England         2.0         1.0   Friendly  Glasgow   
3  1875-03-06   England  Scotland         2.0         2.0   Friendly   London   
4  1876-03-04  Scotland   England         3.0         0.0   Friendly  Glasgow   

    country continent  
0  Scotland    Europe  
1   England    Europe  
2  Scotland    Europe  
3   England    Europe  
4  Scotland    Europe

In [625]:
shootouts.head()

date    home_team         away_team       winner winner_continent  \
0  1967-08-22        India            Taiwan       Taiwan             Asia   
1  1971-11-14  South Korea  Vietnam Republic  South Korea             Asia   
2  1972-05-07  South Korea              Iraq         Iraq             Asia   
3  1972-05-17     Thailand       South Korea  South Korea             Asia   
4  1972-05-19     Thailand          Cambodia     Thailand             Asia   

  away_continent  
0           Asia  
1            NaN  
2           Asia  
3           Asia  
4           Asia

## Summarize Your Results

Make note of your answers to the following questions.

1. Did you find all four types of dirty data in your dataset?
I was able to find all four types of dirty data in my dataset, some of which presented itself once I started reconciling irregular data or adding in supplemental data. I wanted to have the continent for each country listed across the dataframes, but that then presented inconsistent categorization across dataframes (Scotland, England, Wales, and Northern Ireland being listed as United Kingdom in one dataframe, and as individual countries in others)

Since the data spans so far back in the future, it is not entirely suprising that certain datapoints are missing, particularly such particular datapoints as the minute a goal was scored in a game. Thankfully this particular datapoint is not needed for my analysis. 

2. Did the process of cleaning your data give you new insights into your dataset?
Cleaning the data, and in particular eliminating unnecessary data, gave a clearer and more refined understanding of the data I have on my hands and what I can do with it. Will all the data up front it feels like there are endless possibilities, but cleaning and refining the data help hone in on the scope of what I'm trying to evaluate. 

3. Is there anything you would like to make note of when it comes to manipulating the data and making visualizations?
There are so many connections across the 3 dataframes I'll be utilizing (formers_names and continents having been used soley for the data cleaning step) that I'll need to keep in mind and be clear the different names across the sheets for the same datapoint. To be determined if changing the column names to be the same across all sheets for the same datapoint would be beneficial or lend itself to more errors. 

In [616]:
results.head()

date home_team away_team  home_score  away_score tournament     city  \
0  1872-11-30  Scotland   England         0.0         0.0   Friendly  Glasgow   
1  1873-03-08   England  Scotland         4.0         2.0   Friendly   London   
2  1874-03-07  Scotland   England         2.0         1.0   Friendly  Glasgow   
3  1875-03-06   England  Scotland         2.0         2.0   Friendly   London   
4  1876-03-04  Scotland   England         3.0         0.0   Friendly  Glasgow   

    country continent  
0  Scotland    Europe  
1   England    Europe  
2  Scotland    Europe  
3   England    Europe  
4  Scotland    Europe

In [617]:
shootouts.head()

date    home_team         away_team       winner winner_continent  \
0  1967-08-22        India            Taiwan       Taiwan             Asia   
1  1971-11-14  South Korea  Vietnam Republic  South Korea             Asia   
2  1972-05-07  South Korea              Iraq         Iraq             Asia   
3  1972-05-17     Thailand       South Korea  South Korea             Asia   
4  1972-05-19     Thailand          Cambodia     Thailand             Asia   

  away_continent  
0           Asia  
1            NaN  
2           Asia  
3           Asia  
4           Asia

In [618]:
goal_scorers.head()

date  home_team away_team       team            scorer continent
0  1916-07-02      Chile   Uruguay    Uruguay   José Piendibene  Americas
1  1916-07-02      Chile   Uruguay    Uruguay  Isabelino Gradín  Americas
2  1916-07-02      Chile   Uruguay    Uruguay  Isabelino Gradín  Americas
3  1916-07-02      Chile   Uruguay    Uruguay   José Piendibene  Americas
4  1916-07-06  Argentina     Chile  Argentina     Alberto Ohaco  Americas